Communication/Protocols/UCI/Options/SpinOption.cs

```diff
@@ -0,0 +1,40 @@
+namespace Communication.Protocols.UCI
+{
+    class SpinOption : BaseOption
```

Comment:
It is completely not clear to me what SpinOption class exists for. For example, the constructor input is ```int minAllowed, int maxAllowed, int value``` and you check if the value is in range. Could you please explain why we require this class in our case?

Reply:
`SpinOption` is one of the Option types introduced by the UCI Standard
http://wbec-ridderkerk.nl/html/UCIProtocol.html (Search for "* type")

Potential usage examples:
1. A spin option for selecting the target ELO rating that Bot will try to imitate. 
2. Real one: `option name NalimovCache type spin default 1 min 1 max 32`